hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTable.java

```diff
@@ -753,7 +755,13 @@ private void deleteInvalidFilesByPartitions(HoodieEngineContext context, Map<Str
           LOG.info("Deleting invalid data file=" + partitionFilePair);
           // Delete
           try {
-            storage.deleteFile(new StoragePath(partitionFilePair.getValue()));
+            StoragePath pathToDelete = new StoragePath(partitionFilePair.getValue());
+            boolean deletionSuccess = storage.deleteFile(pathToDelete);
+            if (!deletionSuccess && storage.exists(pathToDelete)) {
```

Comment:
do we need to do costly `storage.exists` additionally? Anyway, if file is not found at this point, then it's a no-op in the catch block.

Reply:
only if deletion failed. diff file storage scheme might behave differently. So, if deletion failed for some reason we wanted to check if the file of interest exists and if it exists, then we throw exception. 

So, `exists` is only going to be invoked in the following case
- if additional files were found in markers compared to write statuses
- if deletion failed among the ones thats intended to be deleted during reconciliation. 

So, we should rarely hit this scenario